build: Move CI options to CMake presets

nbolton opened the pull request:
Fixes: #7726

This also updates docs, inline with hacking guide, but I have added a TODO to #7673 to reflect any relevant changes.

sithlord48 commented:
I honestly rather set the cmake options in the clear. The more you hide things the harder it is to follow..

To build we need onlt to run cmake -S. -Bbuild 
everything else set in the preset is constricting ppl to those options. wanna build with ninja on linux TOO BAD the preset sets says you can not.. these over bearing presets are as bad as having config and build scripts. your imposing your environment on others

sithlord48 commented on res/dist/arch/PKGBUILD.in:
```diff
@@ -32,7 +32,7 @@ options=('!debug')
   # By default, `makepkg` will run from the `src` directory, which would
   # only install the binaries, and not the .desktop file, etc. To install
-  # everything, we need to run `make install` with the root Makefile.
+  # everything, we need to set `--prefix`.
```
This change is ok

nbolton replied:
Moved to: #7740

sithlord48 commented on CMakePresets.json:
```diff
@@ -36,97 +30,78 @@
+      "generator": "Ninja",
       "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "Debug"
+        "CMAKE_INSTALL_PREFIX": "/usr"
```
great example something you should not be setting for the same reason we dont set it in the cmake file itself..

nbolton replied:
I don't think you can compare the cmake config against cmake presets. That's exactly why cmake presets exists; to standardize dev env across the team and across environments (local and CI). 

Presets are optional, cmake config is not. You don't have to use it if you don't want to.

sithlord48 reviewed (changes requested):
somethings dont belong here (doc changes)

Id rather remove the presets then try to keep to maintain them even more so since our project is so simple to config we really dont' need this at all .

nbolton commented on README.md:
```diff
@@ -61,55 +61,17 @@ Here are a few ways to join in with the project and get involved:
+## Installing
 
-## How to install (packages)
+Community packages coming soon, for now please use: [Continuous Build](https://github.com/deskflow/deskflow/releases)
```
Ok, will move.

nbolton replied:
Moved to #7673

nbolton commented:
> I honestly rather set the cmake options in the clear. The more you hide things the harder it is to follow..

Using presets isn't about hiding things—it's about standardizing for consistency, especially in environments like CI.

- **CMake config** should have sensible defaults for everyone.
- **CMake presets** should help standardize the dev team's environment.

I can't stress enough how crucial it is to avoid drift between CI and dev environments. Without this standardization, we're seeing the environments diverge, which is becoming a serious headache.

> To build we need onlt to run cmake -S. -Bbuild everything else set in the preset is constricting ppl to those options. 

That’s fine for non-devs, packagers, etc. But as developers, we often need a different setup. I’m not suggesting a one-size-fits-all approach—presets are optional and intended to simplify things if you want to align with CI.

> wanna build with ninja on linux TOO BAD the preset sets says you can not.. these over bearing presets are as bad as having config and build scripts. your imposing your environment on others

Using presets is a choice. I use them because I want to replicate the CI environment. If you don’t care about that, you don’t have to use them—they’re entirely optional.

> Id rather remove the presets then try to keep to maintain them even more so since our project is so simple to config we really dont' need this at all .

I’m definitely against removing them. They save me a lot of time and make me more productive. That said, they are optional, and to show that, I’ve removed them from the documentation and added a `dev-` prefix to each preset name to make their purpose clearer.

nbolton commented:
While there's some good stuff in here (e.g. using cmake for Arch package build), on the whole the idea no longer fits with our philosophy: [Project Config](https://github.com/deskflow/deskflow/wiki/Project-Config)

I'll pick out the good stuff and put it in a new PR.

For context, here's where I made a 360 on my position about presets: https://github.com/deskflow/deskflow/issues/7726

nbolton commented on .github/workflows/ci.yml:
```diff
@@ -364,12 +363,14 @@ jobs:
+        export QT_QPA_PLATFORM=offscreen
         ./build/bin/unittests
-        ./build/bin/integtests
+        ./build/bin/integtests || true
```
Isolated to: #7738

nbolton commented on BUILD.md:
```diff
@@ -11,6 +11,8 @@ wiki page if you have problems.
 
 **1. Dependencies:**
 
+You can either copy/paste the commands for your OS from [`config.yaml`](config.yaml) or use the deps script.
```
Moved to #7673
